### DISCUSSION TITLE
SPIFFSEditor: upgrade from: me-no-dev/ESPAsyncWebServer => mathieucarbou/ESPAsyncWebServer

### DISCUSSION THREAD

**Discussion by hvegh** [mathieucarbou/ESPAsyncWebServer#40] [Q&A]
Dear devs,

I am trying to use an esp32-c3 with the original library. I get crashes when trying to do a wifi scan. 
Maybe this is due to due to single core C3 specs?

```
...
*wm:[2] <- HTTP Root 
*wm:[2] <- HTTP Wifi 
*wm:[2] WiFi Scan SYNC started 
E (53054) task_wdt: Task watchdog got triggered. The following tasks did not reset the watchdog in time:
E (53054) task_wdt:  - async_tcp (CPU 0)
E (53054) task_wdt: Tasks currently running:
E (53054) task_wdt: CPU 0: loopTask
E (53054) task_wdt: Aborting.

abort() was called at PC 0x420263bd on core 0
Core  0 register dump:
...
```
Since the  me-no-dev  is a couple of years old I thought I might upgrade to mathieucarbou branch?
Also I noticed the SPIFFSEditor.h is missing in the new branch.

What would be the best way forward in order to get the newer esp chips like the C3 supported?
Is the SPIFFSEditor.cpp/h functionality separated in different library?

Happy to hear any suggestions...

**Comment by hvegh** [comment]
This is now working:

FIles can be uploaded using **pio run -t uploadfs**


```EditorConfig
[env]
framework = arduino
platform = espressif32
board = esp32dev
monitor_speed = 115200
lib_deps =
    ;https://github.com/me-no-dev/AsyncTCP
    ;https://github.com/me-no-dev/ESPAsyncWebServer
    https://github.com/mathieucarbou/AsyncTCP
    https://github.com/mathieucarbou/ESPAsyncWebServer
    https://github.com/tzapu/WiFiManager#feature_asyncwebserver
    https://github.com/paulo-raca/ArduinoBufferedStreams

[env:esp32c3]
board = airm2m_core_esp32c3
build_flags =
        -D CONFIG_ESP_CONSOLE_UART=1 ; settings for esp32c3 without external uart
        -D ARDUINO_USB_MODE=1
        -D ARDUINO_USB_CDC_ON_BOOT=1
        -D CONFIG_ASYNC_TCP_RUNNING_CORE=1
```

**Comment by mathieucarbou** [comment]
> Also I noticed the SPIFFSEditor.h is missing in the new branch.

SPIFFSEditor is an application on itself which is linked to a specific and fixed UI. 
It is not a server-side component. 
So it was removed from this fork, which is focusing on maintenance of the server part. 
If you need it, just copy/paste the 2-3 files from the original repo in your project, and you should be able to improve it and change the UI like you want also.

Thanks!
